cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/namespace/DefaultNamespaceAdmin.java

```diff
@@ -258,16 +258,16 @@ public synchronized void deleteDatasets(Id.Namespace namespaceId)
 
     if (checkProgramsRunning(namespaceId)) {
       throw new NamespaceCannotBeDeletedException(namespaceId,
-                                                  "Some programs are currently running in namespace " + namespaceId);
+                                                  String.format("Some programs are currently running in namespace " +
+                                                                  "'%s', please stop them before deleting datasets " +
+                                                                  "in the namespace.",
+                                                                namespaceId));
     }
 
     LOG.info("Deleting data in namespace '{}'.", namespaceId);
```

Comment:
OK, but why is this "Deleting data in namespace '{}'." and not deleting datasets?